WordPress/UITestsFoundation/Screens/MySiteScreen.swift

```diff
@@ -17,76 +18,88 @@ private struct ElementStringIDs {
-    let navBar: XCUIElement
-    let removeSiteButton: XCUIElement
-    let removeSiteSheet: XCUIElement
-    let removeSiteAlert: XCUIElement
-    let activityLogButton: XCUIElement
-    let jetpackScanButton: XCUIElement
-    let jetpackBackupButton: XCUIElement
-    let postsButton: XCUIElement
-    let mediaButton: XCUIElement
-    let statsButton: XCUIElement
-    let siteSettingsButton: XCUIElement
-    let createButton: XCUIElement
+
+    let activityLogButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.cells[ElementStringIDs.activityLogButton]
+    }
+
+    let postsButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.cells[ElementStringIDs.postsButton]
+    }
+
+    let mediaButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.cells[ElementStringIDs.mediaButton]
+    }
+
+    var mediaButton: XCUIElement { mediaButtonGetter(app) }
+
+    let statsButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.cells[ElementStringIDs.statsButton]
+    }
+
+    let createButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons[ElementStringIDs.createButton]
+    }
     let readerButton: XCUIElement
-    let switchSiteButton: XCUIElement
+
+    let switchSiteButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons[ElementStringIDs.switchSiteButton]
+    }
 
     static var isVisible: Bool {
         let app = XCUIApplication()
         let blogTable = app.tables[ElementStringIDs.blogTable]
         return blogTable.exists && blogTable.isHittable
     }
 
-    public init() throws {
-        let app = XCUIApplication()
+    public init(app: XCUIApplication = XCUIApplication()) throws {
         tabBar = try TabNavComponent()
-        removeSiteButton = app.cells[ElementStringIDs.removeSiteButton]
-        removeSiteSheet = app.sheets.buttons.element(boundBy: 0)
-        removeSiteAlert = app.alerts.buttons.element(boundBy: 1)
-        activityLogButton = app.cells[ElementStringIDs.activityLogButton]
-        jetpackScanButton = app.cells[ElementStringIDs.jetpackScanButton]
-        jetpackBackupButton = app.cells[ElementStringIDs.jetpackBackupButton]
-        postsButton = app.cells[ElementStringIDs.postsButton]
-        mediaButton = app.cells[ElementStringIDs.mediaButton]
-        statsButton = app.cells[ElementStringIDs.statsButton]
-        siteSettingsButton = app.cells[ElementStringIDs.settingsButton]
-        createButton = app.buttons[ElementStringIDs.createButton]
         readerButton = app.buttons[ElementStringIDs.ReaderButton]
-        switchSiteButton = app.buttons[ElementStringIDs.switchSiteButton]
-        navBar = app.navigationBars[ElementStringIDs.navBarTitle]
 
-        super.init(element: navBar)
+        try super.init(
+            expectedElementGetters: [
+                switchSiteButtonGetter,
+                statsButtonGetter,
+                postsButtonGetter,
+                mediaButtonGetter,
+                createButtonGetter
```

Comment:
Only kept a subset of the elements that were instantiated in the previous version to avoid spending too much time verifying whether the screen is loaded (reminder: this is something `ScreenObject` does automatically at `init` time).